pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -282,3 +275,27 @@ func contextFromChannelAndMaxWaitDuration(stopCh <-chan struct{}, maxWait time.D
 	}()
 	return ctx, cancel
 }
+
+// contextFromChannel returns a Context that is bound to the
+// specified channel. The derived context will be
+// cancelled when the specified channel stopCh is closed.
+//
+// Note the caller must *always* call the CancelFunc, otherwise resources may be leaked.
+func contextFromChannel(stopCh <-chan struct{}) (context.Context, context.CancelFunc) {
```

Comment:
There is already `ContextForChannel` in:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go#L116

Doesn't it do the same?

Reply:
Hmm, this is a copy of what was there when this was written; I do not remember why.

BTW, wait.ContextForChannel has been modified since, I am going to have to rebase.